pkg/integration/handler.go

```diff
@@ -0,0 +1,214 @@
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/dhis2-sre/im-manager/pkg/config"
+
+	"github.com/dhis2-sre/im-manager/internal/handler"
+	"github.com/gin-gonic/gin"
+)
+
+func NewHandler(config config.Config, client DockerHubClient) Handler {
+	return Handler{config, client}
+}
+
+type Handler struct {
+	config          config.Config
+	dockerHubClient DockerHubClient
+}
+
+type DockerHubClient interface {
```

Comment:
Did you write a unit test? If not I would not bother with declaring an interface. You have only one implementation and no unit test.